cocotb/scheduler.py

```diff
@@ -319,13 +320,36 @@ def begin_test(self, trigger=None):
                 self.schedule(test)
                 self.advance()
 
-    def react(self, trigger, depth=0):
-        """React called when a trigger fires.
+    def react(self, trigger):
+        """
+        Called when a trigger fires.
+
+        We ensure that we only start the event loop once, rather than
+        letting it recurse.
+        """
+        if self._is_reacting:
+            # queue up the trigger, the event loop will get to it
+            self._pending_triggers.append(trigger)
+        else:
```

Comment:
An early return would be potentially nicer here, but I leave it up to you.

Reply:
Fine be me either way - changed